illumos-utils/src/running_zone.rs

```diff
@@ -438,7 +446,10 @@ pub struct InstalledZone {
     link: Option<Link>,
 }
 
-impl InstalledZone {
+impl<Port> InstalledZone<Port>
+where
+    Port: OptePort,
+{
```

Comment:
Are these generic parameters + the `OptePort` trait actually necessary? Previously, a snippet of code in `sled-agent/src/opte/mod.rs` conditionally used either the "Meant for illumos" or "not meant for illumos" variant depending on target platform:

https://github.com/oxidecomputer/omicron/blob/e12441e9fcd8529cc6924d90db9ff963e1b1ae52/sled-agent/src/opte/mod.rs#L5-L13

Did you add these to break the `illumos-utils` -> `sled-agent` dependency? This is making me wonder if we should also move the `opte` module to a separate repo...

Regardless, I don't think this needs to hold up this PR, but I'm curious what you think, and if we should file any follow-up issues.

Reply:
> Did you add these to break the illumos-utils -> sled-agent dependency? This is making me wonder if we should also move the opte module to a separate repo...

Precisely this, yes. I'll take a quick stab at moving opte to a separate repo and see how painful that looks.

Reply:
Sorry, I skipped over your first question:

> Are these generic parameters + the OptePort trait actually necessary? 

Not really, _except_ to break the dependency. Everywhere in `sled-agent` that uses `InstalledZone` / `RunningZone` uses the same `Port` generic argument (which is either the "illumos" or "not illumos" version of `Port`, depending on `target_os`), so there's still only one monomorphized instance of those types.

Reply:
Sounds great, thanks for following up on this!